### PR TITLE
[GLUTEN-3715][VL] followup to add CentOS7 support for GCS

### PR DESCRIPTION
**Pull request by zhouyuan**
## What changes were proposed in this pull request?

add GCS support on Centos7, also removed the unnecessary arrow components in static build job

## How was this patch tested?

pass GHA

### CODE REVIEW

File: tools/gluten-te/centos/dockerfile-buildenv [hunk 1]
```diff
@@ -56,13 +56,13 @@ RUN wget https://downloads.apache.org/maven/maven-3/3.8.8/binaries/apache-maven-
     rm apache-maven-3.8.8-bin.tar.gz && \
     mv apache-maven-3.8.8 /usr/lib/maven
 
-# Build & install Spark 3.2.2
-RUN cd /opt && git clone --depth 1 --branch v3.2.2 https://github.com/apache/spark.git spark322
-RUN cd /opt/spark322 && ./build/mvn -Pyarn -DskipTests clean install
+# # Build & install Spark 3.2.2
```
Comment (zhouyuan) [inline tools/gluten-te/centos/dockerfile-buildenv:59]: these are commented out as gluten only run spark unit tests on ubuntu

File: dev/vcpkg/vcpkg.json [hunk 2]
```diff
@@ -2,25 +2,8 @@
   "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
   "builtin-baseline": "a7b6122f6b6504d16d96117336a0562693579933",
   "dependencies": ["jemalloc"],
-  "default-features": ["arrow", "velox", "velox-s3", "velox-gcs", "velox-hdfs"],
+  "default-features": ["velox", "velox-s3", "velox-gcs", "velox-hdfs"],
```
Comment (zhouyuan) [inline dev/vcpkg/vcpkg.json:5]: not necessary now

File: .github/workflows/velox_be.yml [hunk 3]
```diff
@@ -393,23 +388,25 @@ jobs:
...
         run: |
-          docker run --rm --init --privileged --ulimit nofile=65536:65536 --ulimit core=-1 --security-opt seccomp=unconfined \
-          -v $PWD:/opt/gluten --name centos7-test-$GITHUB_RUN_ID -e NUM_THREADS=30 -detach 10.0.2.4:5000/gluten-dev/centos:7 \
-          bash -c 'cd /opt/gluten && sleep 14400'
+          EXTRA_DOCKER_OPTIONS="--name centos7-test-$GITHUB_RUN_ID -e NUM_THREADS=30 --detach" \
+          NON_INTERACTIVE=ON \
+          MOUNT_MAVEN_CACHE=OFF \
+          OS_IMAGE=centos:7 \
+          OS_VERSION=7 \
+          tools/gluten-te/centos/cbash.sh sleep 14400
```
Comment (zhouyuan) [inline .github/workflows/velox_be.yml:396]: using vanilla centos image here as the dependency is broken in pre-built image

**Review by PHILO-HE — APPROVED** [review]
Looks good!

**Pull request merged**